app/_components/useContext/AuthContext.tsx

```diff
@@ -0,0 +1,57 @@
+'use client'
+import { createContext, useEffect, useState } from "react";
+
+import { User, getAuth } from "firebase/auth";
+import { getAuthService } from "@/config/firebaseconfig";
+getAuth
+
+export const AuthContext = createContext<User|null>(null);
+
+interface AuthProviderProps {
+  children: React.ReactNode
+}
+
+export const AuthProvider: React.FC<AuthProviderProps> = ({ children }) => {
+  const [user, setUser] = useState<User | null>(null);
+  
+  useEffect(() => {
+    void handleIsLoggedIn();
+
+    return () => {};
+  },[]);
+
+  const handleIsLoggedIn = async () => {
+    try {
+      const auth = await getAuthService();
+      auth.onAuthStateChanged(user => {
+        try {
+          if (!user) throw 'No user';
+            setUser(user);
+            console.log(user.uid)
+        } catch(error) {
+          console.log(error);
+          setUser(null);
+        }
+      });
+    } catch (error) {
+      console.log(error);
```

Comment:
This should be `console.error`.

In the future, it would be ideal for this error to get handled in some way. Maybe we have an `errors` object that gets updated and displays a helpful message on the FE.